src/models/Collection.ts

```diff
@@ -0,0 +1,198 @@
+      return e
+    }
+  },
+  async addMany(collections: CollectionRecords, db: Context, userId: string) {
+    try {
+      const existing = await CollectionModel.getAll(userId, db)
+      if (existing.length >= 25 || existing.length + collections.length > 25) {
+        return new Error('You have reached the maximum number of collections')
+      }
+    } catch (e) {
+      console.error('Error in CollectionModel.addOne', e)
+    }
+
+    const newCollections = collections.map((collection: CollectionRecord) => ({
+      _id: new ObjectId(),
+      // #TODO Future data modeling to be done for canonical collections
+      cmsId: collection.id,
+      title: collection.title,
+      bookmarks: collection.bookmarks.map((bookmark: BookmarkRecord) => ({
+        _id: new ObjectId(),
+        cmsId: bookmark.id,
+        url: bookmark.url,
+        label: bookmark.label,
+      })),
+    }))
+
+    const query = {
+      userId: userId,
+    }
+
+    const updateDocument = {
+      $push: {
+        mySpace: {
+          $each: newCollections,
+        },
+      },
+    }
+
+    try {
+      await db.collection('users').updateOne(query, updateDocument)
+      const updatedCollections = await db
+        .collection('users')
+        .find({
+          userId: userId,
+        })
+        .toArray()
+
+      return updatedCollections[0].mySpace
+    } catch (e) {
+      // TODO error logging
+      // eslint-disable-next-line no-console
+      console.error('error in add collections', e)
+      return e
+    }
+  },
+  async editOne(_id: string, title: string, db: Context, userId: string) {
+    const query = {
+      userId: userId,
+      'mySpace._id': new ObjectId(_id),
+    }
+
+    const updateDocument = {
+      $set: {
+        'mySpace.$.title': title,
+      },
+    }
+
+    try {
+      await db.collection('users').updateOne(query, updateDocument)
+
+      const updatedCollection = await db
+        .collection('users')
+        .find({ 'mySpace._id': new ObjectId(_id) })
+        .project({ 'mySpace.$': 1, _id: 0 })
+        .toArray()
+
+      return updatedCollection[0].mySpace[0]
+    } catch (e) {
+      // TODO error logging
+      // eslint-disable-next-line no-console
+      console.error('error in edit collection', e)
+      return e
+    }
+  },
+  async deleteOne(_id: string, db: Context, userId: string) {
+    const query = {
+      userId: userId,
+    }
+
+    const updateDocument = {
+      $pull: {
+        mySpace: {
+          _id: new ObjectId(_id),
+        },
+      },
+    }
+
+    try {
+      // Update and save modified document
+      await db
+        .collection('users')
+        .findOneAndUpdate(query, updateDocument, { returnDocument: 'after' })
+
+      // Return deleted collection id
+      return { _id: _id }
+    } catch (e) {
+      // TODO error logging
+      // eslint-disable-next-line no-console
+      console.error('error in remove collection', e)
+      return e
```

Comment:
I noticed a few different patterns for error handling (some places return, some throw, some swallow the error entirely) and I think we need to nail down a consistent pattern here, probably `throw` and ensure errors are surfaced back to the UI (so the user can be notified something went wrong). It would also be nice to establish conventions for the error messages since they'll be in the server logs, maybe something like: `CollectionModel Error: error removing collection`